api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
Sorry, I'm not sure I follow. #24186 says "This is more of an enhancement, but I think in the context of a swarm/cluster, I may want to list the containers participating in the entire network, not just on the current node." Isn't the same thing as adding a `network` selector to `ListTasksRequest`?

It's actually not very difficult to support this, because tasks and services are already indexed by network in swarmkit's data store. #24186 is an old issue from before we were able to do this.